### ISSUE TITLE
LiftGraph.png file has blank data

### ISSUE THREAD

**Issue by DeeEmm** [DeeEmm/DIY-Flow-Bench#157]
Yes it saves liftdata.json and it show values correctly in text format when I open it with notepad, but image LiftGraph.png show 12kb size and 800x750 pix, but when the file is opened it is empty, there is no image, the opened image shows as a transparent rectangle. A simple solution for now is just a screenshot, but probably there is a bug in code, because same problem on phone and pc when image is saved. 
Another small detail that would be convenient is the possibility to change not only the lift interval but also the maximum lift and maximum flow values on the graph in the configuration. Now the graph looks out of proportion with the maximum 1000cfm when measuring for example up to 350cfm.

_Originally posted by @black-top in https://github.com/DeeEmm/DIY-Flow-Bench/discussions/155#discussioncomment-7875544_

**Comment by DeeEmm** [comment]
Just testing this and appears to be working fine.

![LiftGraph (2)](https://github.com/user-attachments/assets/ca4847c4-e170-4e2e-9600-f535474cb372)

@black-top are you still seeing this issue?

Does anyone else see the same problem?

**Comment by black-top** [comment]
Yes, I get a blank image. If I open an image with a different program, I still see the same view. Same on an Android phone.
  
![IMG_20241125_142725](https://github.com/user-attachments/assets/63ba2b35-9c9a-4d36-bc79-587f27b1c6cb)
![IMG_20241125_142555](https://github.com/user-attachments/assets/27270c15-5925-4580-9c60-8b3c1e151770)

**Comment by DeeEmm** [comment]
Can I ask what browser you are using? Looks to be Chrome?

I'm using Chrome / Firefox / Opera here wth no issues but I run a MAC so do not have a way to test Internet Explorer. 

The conversion from the SVG image displayed in the UI to downloadable image happens entirely in the browser. The process takes the SVG image, converts it and presents it the browser as a download. This is done in javascript at runtime.

I suspect that the issue is therefore be browser related. The big question is how / why.

One easy thing to test is to change the file format. I've just committed a change to the DEV branch that changes the file from PNG to JPG. Give that a whirl and see if it is any different.

You will need to compile / upload as well as use the index.html in the data folder.

Let me know if that makes a difference

**Comment by black-top** [comment]
Don't have any time to do testing right now, so tested  only different browsers with same firmware in esp32. It looks like the problem is with Firefox, because at the flow bench win7 pc with Firefox and on an Android phone with Firefox, the problem is the same.  With Opera browser, image saving is working.

**Comment by stefan63** [comment]
Firefox is picky my experience to!
Chrome is restrictive but at least it give you an option ;)

Den lör 30 nov. 2024 kl 18:06 skrev Black-top ***@***.***>:

**Comment by DeeEmm** [comment]
Thanks @black-top. Will take a look at the requirements for the image saving in FF.

I use Chrome.

**Comment by DeeEmm** [comment]
I have replicated this issue locally in FF.

Changing file download type from PNG to JPG did not resolve the issue.

Seems as if the canvas is not being saved properly. (SVG is painted onto HTML canvas and then saved as an image). There are some references to issues with FF and fixes discussed, but it looks like those fixes are already present in the code.

Currently we have no CSS data associated with the canvas as none is needed, but there is at least some commentary online that it is important to integrate the CSS. Perhaps this needs to be present for this work in FF.

Will try and test again.

**Comment by DeeEmm** [comment]
Just re-tested this to work on a fix and this issue now appears to be resolved.

This may possibly be Firefox related - perhaps a recent update. My FF version is V 133.0.3
